server.py

```diff
@@ -0,0 +1,84 @@
+import argparse
+import socket
+import struct
+import sys
+import threading
+THE_UMOUNT_OF_WRONG_CONECTOIN = 5
+THE_UMOUNT_OF_BITES_IN_NUMER_OF_LEN = 4
+
+
+'''
+we make a server that can acssept several clients and read an encoded massege
+'''
+def recv_all (sock, bufsize, flags=0):
+    '''
+    get the massege from client and make it readabol
+    :param sock: the massege from client
+    : type sock: socket.socket
+    :param bufsize: the length of massege from client
+    :type bufsize: int
+    :param flags: the flags in revc
+    :type flags: int
+    :return: the massege from client
+    :rtype: string
+    '''
+    result =b''
+    while (lenght_diff :=bufsize -len(result))>0:
+        result += sock.recv(lenght_diff, flags)
+    return result
+
+
+def run_server (server_ip, server_port):
+    '''
+    read from client and print it, can get from a number of clients
+    :param server_ip: the ip of the server
+    : type server_ip: int
+    :param server_port: the port of the server
+    :type server_port: int
+    '''
+    serv = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    serv.bind((server_ip, server_port))
+    serv.listen(THE_UMOUNT_OF_WRONG_CONECTOIN)
+    while True:
+        conn, addr = serv.accept()
+        t = threading.Thread(target=run_server, args=(server_ip, server_port+1))
+        t.start()
```

Comment:
So you start a new server (with a different port) in a new thread each time?
This is quite odd and not the required behviour.

We want 1 server to run with 1 ip, port and support handling multiple connections.
Notice that in your case, if multiple clients will try to simultaneously some might succeed (the ones that are actually connecting one by the other in this case), and a lot will fail. In addition, multiple new servers will start running, but no client is even trying to connect to them, as they are on completely different ports.